client/blocks/reader-post-card/index.jsx

```diff
@@ -165,8 +169,43 @@ class ReaderPostCard extends React.Component {
 			/>
 		);
 
+		// Set up post byline
+		let postByline;
+
+		if ( isDiscoverStream && ! isEmpty( discoverPick ) ) {
```

Comment:
Just moved. Jan requested that the byline markup live inside the compact post card (https://github.com/Automattic/wp-calypso/pull/16256#issuecomment-316236710), so I rejigged the order so `postByline` can be provided to `CompactPostCard` as a prop.